Update Hist.cxx to implement AdoptErrors()

CiaranH1997 opened the pull request:
Implement function AdoptErrors(), which allows a Hist object to be created with the SumSq array set manually

cjbackhouse commented on CAFAna/Core/Hist.cxx:
```diff
@@ -344,7 +354,7 @@ namespace ana
     fSumSq.resize(0);
     fSqrtErrs = true;
   }
-
+  
```
Undo this accidental whitespace change.

cjbackhouse commented on CAFAna/Core/Hist.cxx:
```diff
@@ -154,6 +154,16 @@ namespace ana
   }
 
+  //----------------------------------------------------------------------
+  Hist Hist::AdoptErrors(Eigen::ArrayXd&& v, Eigen::ArrayXd&& e)
```
I was confused by this name for a moment because somehow I thought it would only set the errors. Maybe `AdoptWithErrors`?

cjbackhouse replied:
Can you include the `.h` change in this same PR so it can all be merged together?

cjbackhouse reviewed (commented):
Presumably for this to be any use to the end user you also need some change to Spectrum? This is fine, but let's see how that part looks, first.

cjbackhouse commented:
One more thought - can we make clear by the argument name and in a comment that this is the array of *squared* errors?

CiaranH1997 commented:
I have changed the name of the function to AdoptWithErrors() and changed the argument name to be sqerr to make it clear these are the errors squared. I did not think this would require change to Spectrum. Spectrum has a constructor which can take a Hist as an argument. So I thought if you have set fSumSq in the Hist and then give this Hist to the Spectrum everything should just work. But maybe I have missed something.

cjbackhouse commented:
I don't remember why the Hist-based constructor exists. Isn't the ArrayXd one as good? I think I'd probably prefer to add a new Spectrum constructor taking two arrays to prevent things outside of Spectrum relying on the Hist implementation derail.
Question: what happens if the environment variable is not set? At what stage will we skip computing the errors? Should it be an error to call this function in that case, or should we silently discard the errors?

CiaranH1997 commented:
I can add in an extra constructor for Spectrum to take two arrays if you think it would be better. Based on one of the comments in Spectrum.h, I got the impression the constructor taking a Hist was meant to be a special case anyway to be used in ReweightableSpectrum. If we don't think this new constructor will be used by anyone other than prism people then I think throwing an error may be better. Don't want future analyzers thinking their statistical uncertainties are correct when they are not.
